src/UI/examples/Card/RepositoryObject/with_object_icon_and_certificate.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/* Copyright (c) 2018 Jesús López <lopez@leifos.com> Extended GPL, see docs/LICENSE */
+
+function with_object_icon_and_certificate() {
+	//Init Factory and Renderer
+	global $DIC;
+	$f = $DIC->ui()->factory();
+	$renderer = $DIC->ui()->renderer();
+
+	$icon = $f->icon()->custom(ilUtil::getImagePath("icon_crs.svg"), 'Course', 'responsive');
```

Comment:
We are implementing this on the base of the custom icons since this is the only way we can assure that we can deliver the feature. We are asked by JF to use the files for the outlined icons, which is perfectly possible using the custom icons. JF did not ask us to introduce a new KS element for outlined icons. Please accept this on the base of custom icons. I am happy if you move this to the new KS outlined icon afterwards.

Reply:
Hi @alex40724 the JF accepted the interface which requires an Icon which is either a standard or custom. There might be cases, where you will need a custom Icon. However this is surely not one of those. Read the description of the custom Icon:

> ILIAS allows users to upload icons for repository objects. Those, in opposite to the standard icons, need to be constructed with a path.

What would be the purpose of having the standard icon if they are not use for the standard icon set?

`We are asked by JF to use the files for the outlined icons, which is perfectly possible using the custom icons. JF did not ask us to introduce a new KS element for outlined icons. `

Nope the JF did not. I did that (for you). The JF accepted my proposal for those oultined Icons on mondey.  Further, note that the outlined Icons are also standard icons, just with an addtional flag set. All you need to do is, make sure, that this works with the standard Icon. If so, the outlined Icons will work as well (since they ARE standard icons and rendered in the exact same manner see: #1228). 

If standard icons mess up the CSS, let me know if you need help. I am sure, we can fix that.

Reply:
Maybe it's also worth to point out again, that we do not need to wait for the merge of the outlined Icons to merge this. I'll do so, as soon as the pointed out issues are fixed. I expect the outlined Icons to be merged in the middle of next week (when @klees is back for the final review and merge and Yvonne has the completed set ready).

Reply:
Am not sure whether we refer to the same JD decision: "24 SEP 2018: We highly appreciate this suggestion. We decidec that icons in the snowboard should not be clickable but only the background image below and the title. We will use the outline icons offered by CaT for presenting the object type and the certificate. For the progress meter we will use the mini version plus text (and without background). We will try to solve the responsive problem of the progress meter as well."

Does it say we must implement a new KS element?

Reply:
ILIAS repository objects allow to upload icons. We only can use custom icons here. As soon as we allow uploading outlined versions for repository objects (maybe with the next release) we will use custom icons anyway, or am I wrong? This is why we focused on making them auto-scalable. This is easy with images. I did not work out a solution for divs with background images here, since I thought it is acceptable to use the custom icons for now.

I also read the JF decision this way.

If you can provide CSS that autoscales the standard icons in any container we are happy to use this.

Reply:
Hi @alex40724 see my comment bellow.